pkg/controller/direct/directbase/operations.go

```diff
@@ -98,6 +99,23 @@ func (o *CreateOperation) RecordUpdatingEvent() {
 	r.Event(o.object, corev1.EventTypeNormal, k8s.Updating, k8s.UpdatingMessage)
 }
 
+func (o *CreateOperation) SetSpecResourceID(ctx context.Context, resourceID string) error {
+	log := klog.FromContext(ctx)
+
+	u := o.GetUnstructured()
+	log.Info("setting resourceID after creation", "resourceID", resourceID)
+
+	if err := unstructured.SetNestedField(u.Object, resourceID, "spec", "resourceID"); err != nil {
```

Comment:
Do you mind adding more context about this "controller writes to spec" logic? We believe we have tried very hard to not do this (even for `.spec.resourceID`).

Reply:
We have the doc about why not use `state-into-spec`, and because some resourceID is server-generated, so that field falls into the same category.  Can I ask what this policy should for? Maybe I miss a broader case. But basically I think if we do not open the door, it does not need explain.

Reply:
I believe this has not  been written down with the docs/ai/server-generated-id.md change.